test/unit/processing/statement/send-statement.test.js

```diff
@@ -0,0 +1,104 @@
+  })
+
+  test('should call updateScheduleByScheduleId', async () => {
+    await sendStatement(scheduleId, statement, mockTransaction)
+    expect(updateScheduleByScheduleId).toHaveBeenCalled()
+  })
+
+  test('should call updateScheduleByScheduleId once', async () => {
+    await sendStatement(scheduleId, statement, mockTransaction)
+    expect(updateScheduleByScheduleId).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call updateScheduleByScheduleId with scheduleId and mockTransaction', async () => {
+    await sendStatement(scheduleId, statement, mockTransaction)
+    expect(updateScheduleByScheduleId).toHaveBeenCalledWith(scheduleId, mockTransaction)
+  })
+
+  test('should call publishStatement', async () => {
+    await sendStatement(scheduleId, statement, mockTransaction)
+    expect(publishStatement).toHaveBeenCalled()
+  })
+
+  test('should call publishStatement once', async () => {
+    await sendStatement(scheduleId, statement, mockTransaction)
+    expect(publishStatement).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call publishStatement with statement', async () => {
+    await sendStatement(scheduleId, statement, mockTransaction)
+    expect(publishStatement).toHaveBeenCalledWith(statement)
+  })
+
+  test('should throw when publishStatement rejects ', async () => {
+    publishStatement.mockRejectedValue(Error)
+    const wrapper = async () => {
+      await sendStatement(scheduleId, statement, mockTransaction)
+    }
+    await expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw error when publishStatement rejects ', async () => {
+    publishStatement.mockRejectedValue(Error)
+    const wrapper = async () => {
+      await sendStatement(scheduleId, statement, mockTransaction)
+    }
+    await expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw error with message "Failed to send statement with scheduleId of {scheduleId}" when publishStatement rejects ', async () => {
+    publishStatement.mockRejectedValue(Error)
+    const wrapper = async () => {
+      await sendStatement(scheduleId, statement, mockTransaction)
+    }
+    await expect(wrapper).rejects.toThrow(new Error(`Failed to send statement with scheduleId of ${scheduleId}`))
+  })
+
+  test('should throw when updateScheduleByScheduleId rejects ', async () => {
+    updateScheduleByScheduleId.mockRejectedValue(Error)
+    const wrapper = async () => {
+      await sendStatement(scheduleId, statement, mockTransaction)
+    }
+    await expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw error when updateScheduleByScheduleId rejects ', async () => {
+    updateScheduleByScheduleId.mockRejectedValue(Error)
+    const wrapper = async () => {
+      await sendStatement(scheduleId, statement, mockTransaction)
+    }
+    await expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw error with message "Failed to send statement with scheduleId of {scheduleId}" when updateScheduleByScheduleId rejects ', async () => {
```

Comment:
For these I tend to address it as "message which starts with, "Failed to send statement with scheduleId" when..."

Then the expect becomes,  `await expect(wrapper).rejects.toThrow(/^Failed to send statement with scheduleId/)` (regex-y to says start with this)

Reason being, if we change what the message logs (to an extent) it doesn't change the functionality